controllers/managedocs_controller.go

```diff
@@ -662,8 +674,38 @@ func (r *ManagedOCSReconciler) reconcileStorageCluster() error {
 }
 
 func (r *ManagedOCSReconciler) getDesiredConvergedStorageCluster() (*ocsv1.StorageCluster, error) {
-	sizeAsString := r.addonParams[storageClassSizeKey]
+	sc, err := r.getDesiredStorageCluster(templates.StorageClusterTemplate.DeepCopy())
+	if err != nil {
+		return nil, err
+	}
+	return sc, nil
+}
 
+func (r *ManagedOCSReconciler) getDesiredProviderStorageCluster() (*ocsv1.StorageCluster, error) {
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.onboardingTicketSecret, func() error {
+		onboardingTicketKeyData := r.addonParams[onboardingTicketKey]
+		if err := r.own(r.onboardingTicketSecret); err != nil {
+			return err
+		}
+		r.onboardingTicketSecret.Data = map[string][]byte{
+			"key": []byte(onboardingTicketKeyData),
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to update onboardingTicketSecret: %v", err)
+	}
+
+	sc, err := r.getDesiredStorageCluster(templates.ProviderStorageClusterTemplate.DeepCopy())
```

Comment:
Same as above, why do we need another helper method? provider and converged should have separate implementations that can evolve separately.  

If you need shared functionality between converged and provider please move the specifics into a util function not the entire reconcile function